pkg/receive/hashring.go

```diff
@@ -236,19 +236,14 @@ func newMultiHashring(algorithm HashringAlgorithm, replicationFactor uint64, cfg
 		cache: make(map[string]Hashring),
 	}
 
-	newHashring := func(endpoints []string) Hashring {
-		switch algorithm {
-		case AlgorithmHashmod:
-			return simpleHashring(endpoints)
-		case AlgorithmKetama:
-			return newKetamaHashring(endpoints, SectionsPerNode, replicationFactor)
-		default:
-			return simpleHashring(endpoints)
-		}
-	}
-
 	for _, h := range cfg {
-		m.hashrings = append(m.hashrings, newHashring(h.Endpoints))
+		var hashring Hashring
+		if h.Algorithm != "" {
+			hashring = newHashring(h.Algorithm, h.Endpoints, replicationFactor)
+		} else {
+			hashring = newHashring(algorithm, h.Endpoints, replicationFactor)
+		}
```

Comment:
The if and else seems redundant as the statements are the same.

Reply:
@yeya24 `h.Algorithm` is the algorithm from the `HashringConfig`, `algorithm` is the global algorithm from the parameter.

From my understanding of the issue, we need a new field `HashringAlgorithm` in the `HashringConfig`. If the `HashringAlgorithm` is set in the `HashringConfig`, we'll use that algorithm for that hashring. Otherwise, we'll use the global `HashringAlgorithm`. Is that right?

Reply:
That's true. I am just thinking about having a var for algorithm and then call the newHashring func only once. The if else is still necessary to modify the var.